File: java-pine/src/main/java/com/tipsy/pine/operators/TransformDispatch.java
package com.tipsy.pine.operators;

import com.tipsy.pine.AbstractOperator;
import com.tipsy.pine.OperatorInput;
import com.tipsy.pine.OperatorOutput;

import java.util.Map;

public class TransformDispatch extends AbstractOperator {
    @Override
    public void init(Map<String, Object> params) {}

    @Override
    public void execute(OperatorInput input, OperatorOutput output) {
        String commonField = commonInput.get(0);
        String itemField = itemOutput.get(0);
        Object val = input.common(commonField);
        for (int i = 0; i < input.itemCount(); i++) {
            output.setItem(i, itemField, val);
        }
    }
}

File: pine-python/pine/operators/transform_dispatch.py
from __future__ import annotations

from pine.cancellation import CancellationToken
from pine.operator import (
    AbstractOperator,
    ConcurrentSafe,
    OperatorInput,
    OperatorOutput,
    OperatorParams,
)


class TransformDispatch(AbstractOperator, ConcurrentSafe):
    def __init__(self):
        pass

    def init(self, params: OperatorParams):
        pass

    def execute(
        self, token: CancellationToken, input_: OperatorInput,
        output: OperatorOutput,
    ) -> None:
        common_field = self.common_input()[0]
        item_field = self.item_output()[0]
        val = input_.common(common_field)
        for i in range(input_.item_count()):
            output.set_item(i, item_field, val)
